### PR TITLE
Fix ESLint offences

### PR DESCRIPTION
**Pull request by idelairre**
# What's in this PR?

Lints files and enforces a bunch of rules.

## References
Progress on: #594

### CODE REVIEW

**Comment by idelairre** [comment]
I've also finished the 'no-direct-property-access' but some tests are failing. Hopefully I'll have that up later tonight.

File: mirage/config.js [hunk 1]
```diff
@@ -464,19 +464,19 @@ export default function() {
...
   this.post('/user-roles');
 
   // DELETE /user-roles
   this.delete('/user-roles/:id');
 
-  //////////////
+  // ////////////
   // User skills
-  //////////////
+  // ////////////
```
Comment (JoshSmith) [inline mirage/config.js]: These are rather awkward changes. Should we be doing this another way altogether?

File: tests/integration/components/editor-with-preview-test.js [hunk 2]
```diff
@@ -159,7 +159,7 @@ test('it sets the editor min-height to the editor height when previewing and sti
 
   this.$('.preview').click();
 
-  let style = 'min-height: ' + height + ';';
+  let style = `min-height: ${  height  };`;
```
Comment (JoshSmith) [inline tests/integration/components/editor-with-preview-test.js]: It looks like many of these replacements were made with lots of spacing left over. Not sure if intentional.

File: app/components/scroll-top.js [hunk 3]
```diff
@@ -2,6 +2,6 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
   click() {
-    window.scrollTo(0,0);
+    window.scrollTo(0, 0);
```
Comment (JoshSmith) [inline app/components/scroll-top.js]: This is a weird case.

File: mirage/config.js (hunk 1 above)
Comment (idelairre) [inline mirage/config.js]: This was definitely from using --fix on eslint. I should have given that a closer look. Maybe we should use a different way of making comment blocks stand out, e.g.

```
/**
* User Skills
*/
```
or
```
/*----------------------*
* User Skills
*------------------------/
```
Or we could take out the rule enforce it softly by writing it into a style guide.

File: tests/integration/components/editor-with-preview-test.js (hunk 2 above)
Comment (idelairre) [inline tests/integration/components/editor-with-preview-test.js]: That is a little strange looking, again, probably a result of using --fix. I can hunt for weird spacing in the next commit.

**Comment by idelairre** [comment]
I'm copying the comment I left in the issue thread here.

> The only things left are no-console, camel-case, new-cap, and require-access-in-comments.
> 
> A few notes:
> 
> - I think enforcing camel-case would break the app since the snake-cased properties are coming from the api
> - I wasn't comfortable deleting the console.error entries since one or two of them had comments
> - new-cap was throwing because of jQuery in one test. I'm sure there is a clever way around this or maybe we could just ignore that rule in those instances
> - I have to study the spec for require-access-in-comments to understand how to document correctly, then I can have a go at it.

**Comment by idelairre** [comment]
@JoshSmith Is this good to go or should I add one more commit to fix the funky spacing and comments?

**Comment by JoshSmith** [comment]
I think fix then good!

**Comment by JoshSmith** [comment]
@idelairre not sure why that's causing test failures...

**Comment by idelairre** [comment]
Yeah, that was freaking me out. I couldn't figure out the issue so I redid the commit fixing the spacing on the templates, seems to be ok now.

**Comment by JoshSmith** [comment]
Okay, great. Can you rebase and squash now? I think these are actually too many commits to merge individually but can bundle it all up in one commit message.

**Comment by JoshSmith** [comment]
🙌

**Pull request merged**